### PR TITLE
Slighty bigger than small cleanup of the API

### PR DESCRIPTION
**Pull request by JasonHughes94**
Heyo, just a wee clean up of the Modix Web project.

# Changes
- Add `APIController` attribute to controllers that use `[FromBody]/[FromQuery]` as these are now inferred automatically.
- Update method names to follow style guide.
- Update fields for DI to use private readonly fields over private `{ get; }` properties.
- Remove unused StringULongConvertor.
- Remove RowBoatInfractionsImporter.

### CODE REVIEW

**Comment by JasonHughes94** [comment]
Reverted the namespace change as per discussion in voice

File: Modix/Controllers/ChannelController.cs [hunk 1]
```diff
@@ -34,14 +35,14 @@ public async Task<IActionResult> ChannelDesignations()
...
-        public async Task<IActionResult> RemoveDesignation(long id)
+        public async Task<IActionResult> RemoveDesignationAsync(long id)
         {
-            await ChannelService.RemoveDesignatedChannelByIdAsync(id);
+            await _channelService.RemoveDesignatedChannelByIdAsync(id);
             return Ok();
         }
 
         [HttpPut]
-        public async Task<IActionResult> CreateDesignation([FromBody] DesignatedChannelCreationData creationData)
```
Comment (JakenVeina) [inline Modix/Controllers/ChannelController.cs]: `[FromBody]` is not necessary?

File: Modix/Extensions/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -44,7 +41,7 @@
 using Modix.Services.Wikipedia;
 using StatsdClient;
 
-namespace Microsoft.Extensions.DependencyInjection
+namespace Modix.Extensions
```
Comment (JakenVeina) [inline Modix/Extensions/ServiceCollectionExtensions.cs]: Missed one in your revert.

File: Modix/Mappings/FilterParametersToInfractionSearchCriteriaMapper.cs [hunk 3]
```diff
@@ -8,7 +8,7 @@ public static class FilterParametersToInfractionSearchCriteriaMapper
     {
         public static InfractionSearchCriteria ToInfractionSearchCriteria(this FilterParameter[] filterParameters)
         {
-            return new InfractionSearchCriteria()
+            return new InfractionSearchCriteria
```
Comment (JakenVeina) [inline Modix/Mappings/FilterParametersToInfractionSearchCriteriaMapper.cs]: Personally, I prefer always having `()` for anything that executes code (except properties, of course). Anyone else?

File: Modix/StringULongConverter.cs [hunk 4]
```diff
@@ -1,25 +0,0 @@
-﻿using System;
-using Newtonsoft.Json;
-
-namespace Modix
-{
-    public class StringULongConverter : JsonConverter
```
Comment (JakenVeina) [inline Modix/StringULongConverter.cs]: Was this only used by the Rowboard importer? Or is it just obsolete? I seem to recall @jmazouri being the one who added this, or at least knowing why it was added. There was some particular issue that this was a workaround for.

File: Modix/Mappings/FilterParametersToInfractionSearchCriteriaMapper.cs (hunk 3 above)
Comment (JasonHughes94) [reply Modix/Mappings/FilterParametersToInfractionSearchCriteriaMapper.cs]: I'm 6 and half a dozen about this. Don't mind either way to be honest

File: Modix/Controllers/ChannelController.cs (hunk 1 above)
Comment (JasonHughes94) [reply Modix/Controllers/ChannelController.cs]: Not anymore with the addition of the `[ApiControler]` attribute. Handles binding, and Model State validation for you now. See - https://www.strathweb.com/2018/02/exploring-the-apicontrollerattribute-and-its-features-for-asp-net-core-mvc-2-1/

File: Modix/StringULongConverter.cs (hunk 4 above)
Comment (JasonHughes94) [reply Modix/StringULongConverter.cs]: From what I can tell it wasn't used at all. Not even in the Rowboat importer

File: Modix/Mappings/FilterParametersToInfractionSearchCriteriaMapper.cs (hunk 3 above)
Comment (Sutarodo) [reply Modix/Mappings/FilterParametersToInfractionSearchCriteriaMapper.cs]: If you're not passing in anything to the constructor and you're using field initialisation it's entirely redundant.

Comment (JakenVeina) [reply Modix/Mappings/FilterParametersToInfractionSearchCriteriaMapper.cs]: Sure, it's redundant. It just feels icky.

File: Modix/StringULongConverter.cs (hunk 4 above)
Comment (jmazouri) [reply Modix/StringULongConverter.cs]: This was for cases where ulong IDs were sent as numbers in json, and in clientside JavaScript, they were treated as floating point numbers and lost precision.

I'm not sure when it became unused, but if it's no longer an issue, this isn't necessary.

Comment (jmazouri) [reply Modix/StringULongConverter.cs]: This was for cases where ulong IDs were sent as numbers in json, and in clientside JavaScript, they were treated as floating point numbers and lost precision.

I'm not sure when it became unused, but if it's no longer an issue, this isn't necessary.

**Comment by JasonHughes94** [comment]
Just leaving a wee comment on this to say I will be looking at the bugs @Scott-Caldwell raised over the weekend. I've had a pretty busy week and not had the time at the moment.

**Comment by JasonHughes94** [comment]
Sorry guys, I'm gonna go ahead and close this for the time being. I don't have the time to look into the bugs that Scott raised.

If anyone else wants to pick this up to finish it off feel free.

**Pull request closed without merging**